src/main/java/org/spongepowered/api/text/Texts.java

```diff
@@ -234,6 +251,63 @@ public static Text of(Object... objects) {
             }
         }
 
+        if (builder.children.size() == 1) {
+            // Single content, reduce Text depth
+            return builder.children.get(0);
+        }
+        return builder.build();
+    }
+
+    /**
+     * Builds a {@link Text} from a given array of content objects. This method
+     * assumes, that all elements of the array are content and not formatting.
+     *
+     * <p>This will create the correct {@link Text} instance if the input object
+     * is the input for one of the {@link Text} types or convert the object to a
+     * string otherwise.</p>
+     *
+     * @param separator The separator to be inserted between elements
+     * @param objects The object array
+     * @return The built text object
+     */
+    public static Text joinOf(Text separator, Object... objects) {
+        TextBuilder builder = builder();
+        boolean first = true;
+
+        for (Object part : objects) {
+            if (!first) {
+                builder.append(separator);
+            }
+            first = false;
+            builder.append(Texts.of(part));
```

Comment:
If you're writing a list and list items contain commas, use semicolons for the list instead. (Is my understanding of grammar anyway...)

```
The teams are Player1, Player2, and Player3; Player4, Player5, and Player6; and Player7, Player8, and Player9.